build_tools/build_detect_platform

```diff
@@ -722,8 +745,27 @@ $CXX $PLATFORM_CXXFLAGS $COMMON_FLAGS $TRY_SSE42 -x c++ - -o /dev/null 2>/dev/nu
 EOF
 if [ "$?" = 0 ]; then
   COMMON_FLAGS="$COMMON_FLAGS $TRY_SSE42 -DHAVE_SSE42"
-elif test "$USE_SSE"; then
-  echo "warning: USE_SSE specified but compiler could not use SSE intrinsics, disabling" >&2
+else
+  # SSE4.2 is not available
+  if test "$USE_SSE"; then
+    echo "warning: USE_SSE specified but compiler could not use SSE42 intrinsics, disabling" >&2
+  fi
+
+  # Check for SSE2 (IPPCP requires SSE2 and above)
+  $CXX $PLATFORM_CXXFLAGS $COMMON_FLAGS -x c++ - -o /dev/null 2>/dev/null <<EOF
+    #include <cstdint>
+    #include <emmintrin.h>
+    int main() {
+      const auto x = _mm_set_epi32(0, 0, 0, 0);
```

Comment:
I would just roll this into the test for enabling IPPCP and don't bother with a HAVE_SSE2 flag. Unnecessary complexity IMHO.

Reply:
@pdillinger that is true. Now fixed.